src/firefly/js/visualize/ui/CatalogConstraintsPanel.jsx

```diff
@@ -0,0 +1,441 @@
+                                           labelWidth={75}
+                                           label='Table Selection:'
+                                           multiple={false}
+                        />
+                        <button style={{padding:'0 5px 0 5px', margin:'0 10px 0 10px'}}
+                                onClick={ () => this.fetchDD(catname, dd_short, true)}>Reset
+                        </button>
+                    </div>
+                    <div>
+                        <TablePanelConnected {...{tableModel}} />
+                        {renderSqlArea()}
+                    </div>
+                </div>
+            </div>
+        );
+    }
+
+    /**
+     * Getting dd info from catalog name catName
+     * @param catName string name of the catalog for searching DD information
+     * @param dd_short
+     * @param clearSelections
+     */
+    fetchDD(catName, dd_short, clearSelections = false) {
+        const shortdd = (dd_short == 'true') ? 'short' : 'long';
+        const tblid = `${catName}-${shortdd}-dd-table-constraint`;
+
+        //// Check if it exists already - fieldgroup has a keepState property but
+        //// here we are not using the table as a fieldgroup per se so we need to cache the column restrictions changes
+        const tbl = getTblById(tblid);
+        if (tbl && !clearSelections) {
+            this.setState({tableModel: tbl});
+            return;
+        }
+
+        //const catName = get(FieldGroupUtils.getGroupFields(gkey), 'cattable.value', '');
+        const request = {id: 'GatorDD', 'catalog': catName, 'short': dd_short}; //Fetch DD master table
+        const urldef = get(FieldGroupUtils.getGroupFields(this.props.groupKey), 'cattable.coldef', 'null');
+
+        doFetchTable(request).then((tableModel) => {
+            const tableModelFetched = tableModel;
+            tableModelFetched.tbl_id = tblid;
+            addConstraintColumn(tableModelFetched);
+            addColumnDef(tableModelFetched, urldef);
+            //hideColumns(tableModelFetched);
+            setRowsChecked(tableModelFetched);
+            if (clearSelections) {
+                TblCntlr.dispatchTableReplace(tableModel);
+            }
+            this.setState({tableModel: tableModelFetched});
+        }).catch((reason) => {
+                console.error(reason);
+            }
+        );
+    }
+}
+
+/**
+ * Set true checkboxes column wherever a row should be selected given by column named 'sel'
+ * @param tableModel
+ */
+function setRowsChecked(tableModel) {
+    const selectInfoCls = SelectInfo.newInstance({rowCount: tableModel.totalRows});
+    const idxColSel = getColumnIdx(tableModel, 'sel');
+    tableModel.tableData.data.forEach((arow, index) => {
+        if (arow[idxColSel] === 'y') {
+            selectInfoCls.setRowSelect(index, true);
+        }
+    });
+    const selectInfo = selectInfoCls.data;
+    merge(tableModel, {selectInfo});
```

Comment:
this is a recursive merge.  i don't think you want to recursively merge selectInfo.  use Object.assign or WebUtil.clone instead.
also, it directly modifies the tableModel parameter.  if tableModel is coming from the redux store, it should not be modified.